Initialize Go modules for pulumi/pulumi

jen20 opened the pull request:
This pull request is an _experiment_ with whether or not we can now easily adopt Go modules for dependency management without extensive changes to build infrastructure.

jen20 commented:
I've marked this as "Ready for Review" to get Travis CI to run on it as well as AppVeyor. Please _do not merge_ at this point.

ellismg commented on pkg/backend/display/jsonmessage.go:
```diff
@@ -83,7 +83,7 @@ func (jm *Progress) Display(out io.Writer, termInfo termInfo) {
 		clearLine(out, termInfo)
 		endl = "\r"
-		fmt.Fprintf(out, endl)
+		fmt.Print(out, endl)
```
I don't think this change is correct.

jen20 replied:
You are correct, I’ll fix up.

ellismg commented on pkg/backend/display/progress.go:
```diff
@@ -1224,7 +1225,7 @@ func (display *ProgressDisplay) getStepInProgressDescription(step engine.StepEve
 }
 
-func writeString(b *bytes.Buffer, s string) {
+func writeString(b io.StringWriter, s string) {
```
Just guessing this is due to the newer compiler helpfully telling us to make this change?

jen20 replied:
I think it’s the linter upgrade possibly combined with the compiler upgrade.

jen20 commented:
I've rebased this, updated the modules as per the previous head of `master`, and fixed up the incorrect `Print -> Fprint` change. Pending CI, I think this is good to go.

jen20 commented:
Thanks for looking over this @ellismg! The changes you requested are made and things are green now, so I'm going to go ahead and merge this, and start filtering the changes downstream.
